internal/handlers/internal/gateway/tenant_secrets_test.go

```diff
@@ -0,0 +1,214 @@
+			Name:      "my-stack",
+			Namespace: "some-ns",
+		},
+	}
+
+	s := &lokiv1beta1.LokiStack{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mystack",
+			Namespace: "some-ns",
+		},
+		Spec: lokiv1beta1.LokiStackSpec{
+			Tenants: &lokiv1beta1.TenantsSpec{
+				Mode: lokiv1beta1.Static,
+				Authentication: []lokiv1beta1.AuthenticationSpec{
+					{
+						TenantName: "test",
+						TenantID:   "test",
+						OIDC: &lokiv1beta1.OIDCSpec{
+							Secret: &lokiv1beta1.TenantSecretSpec{
+								Name: "test",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	k.GetStub = func(_ context.Context, name types.NamespacedName, object client.Object) error {
+		if name.Name == "test" && name.Namespace == "some-ns" {
+			k.SetClientObject(object, &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "some-ns",
+				},
+				Data: map[string][]byte{
+					"clientID":     []byte("test"),
+					"clientSecret": []byte("test"),
+					"issuerCAPath": []byte("/path/to/ca/file"),
+				},
+			})
+		}
+		return nil
+	}
+
+	ts, err := GetTenantSecrets(context.TODO(), k, r, scheme, s)
+	require.NoError(t, err)
+
+	expected := []*manifests.TenantSecrets{
+		{
+			TenantName:   "test",
+			ClientID:     "test",
+			ClientSecret: "test",
+			IssuerCAPath: "/path/to/ca/file",
+		},
+	}
+	require.ElementsMatch(t, ts, expected)
+}
+
+func Test_GetTenantSecrets_DynamicMode(t *testing.T) {
+	k := &k8sfakes.FakeClient{}
+	r := ctrl.Request{
+		NamespacedName: types.NamespacedName{
+			Name:      "my-stack",
+			Namespace: "some-ns",
+		},
+	}
+
+	s := &lokiv1beta1.LokiStack{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "mystack",
+			Namespace: "some-ns",
+		},
+		Spec: lokiv1beta1.LokiStackSpec{
+			Tenants: &lokiv1beta1.TenantsSpec{
+				Mode: lokiv1beta1.Dynamic,
+				Authentication: []lokiv1beta1.AuthenticationSpec{
+					{
+						TenantName: "test",
+						TenantID:   "test",
+						OIDC: &lokiv1beta1.OIDCSpec{
+							Secret: &lokiv1beta1.TenantSecretSpec{
+								Name: "test",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	k.GetStub = func(_ context.Context, name types.NamespacedName, object client.Object) error {
+		if name.Name == "test" && name.Namespace == "some-ns" {
+			k.SetClientObject(object, &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test",
+					Namespace: "some-ns",
+				},
+				Data: map[string][]byte{
+					"clientID":     []byte("test"),
+					"clientSecret": []byte("test"),
+					"issuerCAPath": []byte("/path/to/ca/file"),
+				},
+			})
+		}
+		return nil
+	}
+
+	ts, err := GetTenantSecrets(context.TODO(), k, r, scheme, s)
+	require.NoError(t, err)
+
+	expected := []*manifests.TenantSecrets{
+		{
+			TenantName:   "test",
+			ClientID:     "test",
+			ClientSecret: "test",
+			IssuerCAPath: "/path/to/ca/file",
+		},
+	}
+	require.ElementsMatch(t, ts, expected)
+}
+
+func Test_GetTenantSecrets_OpenShiftLoggingMode(t *testing.T) {
```

Comment:
```suggestion
func TestGetTenantSecrets_OpenShiftLoggingMode(t *testing.T) {
```
For consistency in naming everywhere.